modules/mini-games/tightspot.lua

```diff
@@ -0,0 +1,743 @@
+    end
+    for i, player in ipairs(game.connected_players) do
+        player.set_controller {type = defines.controllers.spectator}
+
+        local Main_gui = Gui.get_left_element(player, game_gui)
+        local table = Main_gui.container["Money"].table
+        local Debt = tonumber(table["Debt"].caption) / 5000 -- amount of loans taken
+        Debt = Debt * variables.loan_price --Each loan will cost variables.loan_prices points
+        Store.set(balances, player, Debt * -1)
+        table["Debt"].caption = "0"
+        table["Time"].caption = SecondsToClock(variables.level.play_time / 60)
+
+        if Roles.player_allowed(player, "gui/tightspot_speed") then
+            table = Main_gui.container["slider"].table
+            table.visible = true
+        end
+    end
+    for i, chest in ipairs(chests) do
+        local ent = chest[1]
+        local inv = ent.get_inventory(defines.inventory.chest)
+        inv.clear()
+    end
+end
+
+local function check_chest()
+    if started[1] == true then
+        for i, chest in ipairs(chests) do
+            local ent = chest[1]
+            local inv = ent.get_inventory(defines.inventory.chest)
+            local count = inv.get_item_count(variables.level.demand.item)
+            local price = count * variables.level.demand.price
+            if price ~= 0 then
+                local player = game.players[chest[2]]
+                local balance = Store.get(balances, player)
+                local result = balance + price
+                player.surface.create_entity {
+                    name = "flying-text",
+                    position = ent.position,
+                    text = "+" .. price,
+                    color = {g = 1}
+                }
+                Store.set(balances, player, result)
+            end
+            inv.clear()
+        end
+    end
+end
+
+local function timer()
+    if started[1] ~= true then
+        local time = variables.level.time
+        time = SecondsToClock((time - (game.tick - variables.tick)) / 60)
+        if time == "00:00" then
+            started[1] = true
+            variables.tick = game.tick
+            start_game()
+        else
+            for i, player in ipairs(game.connected_players) do
+                local Main_gui = Gui.get_left_element(player, game_gui)
+                local table = Main_gui.container["Money"].table
+                table["Time"].caption = time
+            end
+        end
+    else
+        local time = variables.level.play_time
+        time = SecondsToClock((time - (game.tick - variables.tick)) / 60)
+        if time == "00:00" then
+            Mini_games.stop_game()
+        end
+        for i, player in ipairs(game.connected_players) do
+            local Main_gui = Gui.get_left_element(player, game_gui)
+            local table = Main_gui.container["Money"].table
+            table["Time"].caption = time
+        end
+    end
+end
+
+token_for_replace_wall = Token.register(replace_wall)
```

Comment:
Why is this so far away from where it is used?

Reply:
idk lol (fixed